clang/lib/Sema/SemaInit.cpp

```diff
@@ -1938,6 +1984,29 @@ static bool checkDestructorReference(QualType ElementType, SourceLocation Loc,
   return SemaRef.DiagnoseUseOfDecl(Destructor, Loc);
 }
 
+static bool canInitializeArrayWithEmbedDataString(ArrayRef<Expr *> ExprList,
+                                                  QualType InitType,
+                                                  ASTContext &Context) {
+  // Only one initializer, it's an embed and the types match;
+  EmbedExpr *First =
+      ExprList.size() == 1
+          ? dyn_cast_if_present<EmbedExpr>(ExprList[0]->IgnoreParens())
+          : nullptr;
+  if (First) {
```

Comment:
First is a weird / unspecific name here.
Also, you could return early if first is null

Reply:
Done, and changed `First` to `EE` (as `EmbedExpr`).